pithos/util.py

```diff
@@ -252,3 +76,12 @@ def is_flatpak() -> bool:
         _is_flatpak = os.path.exists('/.flatpak-info')
 
     return _is_flatpak
+
+_is_msys2 = None
+def is_msys2() -> bool:
+    global _is_msys2
+
+    if _is_msys2 is None:
+        _is_msys2 = sysconfig.get_platform().startswith("mingw")
+
+    return _is_msys2
```

Comment:
I don't believe this does any blocking IO or anything, you can just simplify it down to:

```python
def is_msys2() -> bool:
    return sysconfig.get_platform().startswith("mingw")
```

The caching on Linux is because it access the filesystem (but honestly probably doesn't matter in Python much anyway).